pkg/rules/service.go

```diff
@@ -1,163 +1,80 @@
-func init() {
-	appConfigTemplate = template.Must(template.New("alloy-rules.yaml").Funcs(sprig.FuncMap()).Parse(appConfig))
-}
-
 type Service struct {
-	Client          client.Client
-	AlloyAppVersion semver.Version
+	Client client.Client
 }
 
-func (s *Service) Configure(ctx context.Context) error {
+func (s *Service) Delete(ctx context.Context) error {
 	logger := log.FromContext(ctx)
-	logger.Info("configuring alloy-rules")
+	logger.Info("deleting alloy-rules")
 
-	logger.Info("create or update alloy rules configmap")
-	err := s.createOrUpdateConfigMap(ctx)
+	logger.Info("delete alloy rules app")
+	err := s.deleteApp(ctx)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	logger.Info("configure alloy rules app")
-	err = s.configureApp(ctx)
+	logger.Info("delete alloy rules configmap")
+	err = s.deleteConfigMap(ctx)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	logger.Info("alloy-rules is configured")
+	logger.Info("alloy-rules is deleted")
 
 	return nil
 }
 
-func (s Service) createOrUpdateConfigMap(ctx context.Context) error {
+func (s Service) deleteConfigMap(ctx context.Context) error {
 	logger := log.FromContext(ctx)
-	// Get list of tenants
-	var tenants []string
-	tenants, err := tenancy.ListTenants(ctx, s.Client)
-	if err != nil {
-		return errors.WithStack(err)
-	}
 
 	configmap := &v1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      alloyRulesConfigMapName,
 			Namespace: alloyRulesAppNamespace,
-			Labels:    labels.Common,
 		},
 	}
 
-	_, err = controllerutil.CreateOrUpdate(ctx, s.Client, configmap, func() error {
-		values, err := s.generateAlloyConfig(ctx, tenants)
-		if err != nil {
-			logger.Error(err, "failed to generate %s", alloyRulesConfigMapName)
-			return errors.WithStack(err)
-		}
-
-		data := make(map[string]string)
-		data["values"] = values
-
-		configmap.Data = data
-
-		return nil
-	})
-
-	if err != nil {
-		logger.Error(err, "failed to create or update configmap %s", alloyRulesConfigMapName)
+	err := s.Client.Delete(ctx, configmap)
+	if client.IgnoreNotFound(err) != nil {
+		logger.Error(err, "failed to delete configmap", "configmap", alloyRulesConfigMapName)
 		return errors.WithStack(err)
 	}
-
 	return nil
 }
 
-func (s *Service) generateAlloyConfig(ctx context.Context, tenants []string) (string, error) {
-	data := struct {
-		Tenants []string
-	}{
-		Tenants: tenants,
-	}
-
-	var values bytes.Buffer
-	err := appConfigTemplate.Execute(&values, data)
-	if err != nil {
-		return "", err
-	}
-	return values.String(), nil
-}
-
-func (s Service) configureApp(ctx context.Context) error {
+func (s Service) deleteApp(ctx context.Context) error {
 	logger := log.FromContext(ctx)
 
-	labels := maps.Clone(labels.Common)
-	labels["app-operator.giantswarm.io/version"] = "0.0.0"
 	app := &appv1.App{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      AlloyRulesAppName,
+			Name:      alloyRulesAppName,
 			Namespace: alloyRulesAppNamespace,
-			Labels:    labels,
```

Comment:
Can we keep the labels, just to be sure we are deleting the exact same object here ?

Reply:
I mean,client.Delete only cares abour Name and namespace but sure :)